templates/tfengine/recipes/app_foundation.hcl

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: This example is still under development and is not considered stable!
+# Breaking changes to components for this example will not be part of releases.
+
+template "app" {
+  recipe_path = "./app.hcl"
+  data = {
+    deployments = [
+      {{if has .res "gke_clusters"}}
+      {
+        name = "project_networks",
+        resources = {
+          project = {
+            name_suffix        = "networks"
+            is_shared_vpc_host = true
+            apis               = ["compute.googleapis.com"]
+          }
+          compute_networks = [{
+            name = "primary"
```

Comment:
Is it desirable to have all clusters in the same network by default? Not saying it isn't, but I wanted to get your thoughts.